tests/testthat/test-object_name_linter.R

```diff
@@ -3,38 +3,38 @@ context("object_name_linter")
+  expect_equivalent(lapply(styles, check_style, nms = "X.Y"), list(FALSE, FALSE, FALSE, FALSE, FALSE, FALSE,  FALSE, FALSE))
+  expect_equivalent(lapply(styles, check_style, nms = "x_2"), list(FALSE, FALSE, FALSE,  TRUE, FALSE, FALSE,  FALSE, FALSE))
+  expect_equivalent(lapply(styles, check_style, nms = "X_2"), list(FALSE, FALSE, FALSE, FALSE,  TRUE, FALSE,  FALSE, FALSE))
+  expect_equivalent(lapply(styles, check_style, nms = "x.2"), list(FALSE, FALSE, FALSE, FALSE, FALSE,  TRUE,  FALSE, FALSE))
+  expect_equivalent(lapply(styles, check_style, nms = "X.2"), list(FALSE, FALSE, FALSE, FALSE, FALSE, FALSE,  FALSE, FALSE))
 
 
-  #                                                                       UpC   lowC   snake  SNAKE  dot    alllow  ALLUP
-  expect_equivalent(lapply(styles, check_style, nms = "IHave1Cat"    ), c( TRUE, FALSE, FALSE, FALSE, FALSE,  FALSE, FALSE))
-  expect_equivalent(lapply(styles, check_style, nms = "iHave1Cat"    ), c(FALSE,  TRUE, FALSE, FALSE, FALSE,  FALSE, FALSE))
-  expect_equivalent(lapply(styles, check_style, nms = "i_have_1_cat" ), c(FALSE, FALSE,  TRUE, FALSE, FALSE,  FALSE, FALSE))
-  expect_equivalent(lapply(styles, check_style, nms = "I_HAVE_1_CAT" ), c(FALSE, FALSE, FALSE,  TRUE, FALSE,  FALSE, FALSE))
-  expect_equivalent(lapply(styles, check_style, nms = "i.have.1.cat" ), c(FALSE, FALSE, FALSE, FALSE,  TRUE,  FALSE, FALSE))
-  expect_equivalent(lapply(styles, check_style, nms = "ihave1cat"    ), c(FALSE,  TRUE,  TRUE, FALSE,  TRUE,   TRUE, FALSE))
-  expect_equivalent(lapply(styles, check_style, nms = "IHAVE1CAT"    ), c( TRUE, FALSE, FALSE,  TRUE, FALSE,  FALSE,  TRUE))
-  expect_equivalent(lapply(styles, check_style, nms = "I.HAVE_ONECAT"), c(FALSE, FALSE, FALSE, FALSE, FALSE,  FALSE, FALSE))
+  #                                                                       symbl   UpC   lowC   snake  SNAKE  dot    alllow  ALLUP
+  expect_equivalent(lapply(styles, check_style, nms = "IHave1Cat"    ), c(FALSE,  TRUE, FALSE, FALSE, FALSE, FALSE,  FALSE, FALSE))
+  expect_equivalent(lapply(styles, check_style, nms = "iHave1Cat"    ), c(FALSE, FALSE,  TRUE, FALSE, FALSE, FALSE,  FALSE, FALSE))
+  expect_equivalent(lapply(styles, check_style, nms = "i_have_1_cat" ), c(FALSE, FALSE, FALSE,  TRUE, FALSE, FALSE,  FALSE, FALSE))
+  expect_equivalent(lapply(styles, check_style, nms = "I_HAVE_1_CAT" ), c(FALSE, FALSE, FALSE, FALSE,  TRUE, FALSE,  FALSE, FALSE))
+  expect_equivalent(lapply(styles, check_style, nms = "i.have.1.cat" ), c(FALSE, FALSE, FALSE, FALSE, FALSE,  TRUE,  FALSE, FALSE))
+  expect_equivalent(lapply(styles, check_style, nms = "ihave1cat"    ), c(FALSE, FALSE,  TRUE,  TRUE, FALSE,  TRUE,   TRUE, FALSE))
+  expect_equivalent(lapply(styles, check_style, nms = "IHAVE1CAT"    ), c(FALSE,  TRUE, FALSE, FALSE,  TRUE, FALSE,  FALSE,  TRUE))
+  expect_equivalent(lapply(styles, check_style, nms = "I.HAVE_ONECAT"), c(FALSE, FALSE, FALSE, FALSE, FALSE, FALSE,  FALSE, FALSE))
 })
```

Comment:
Can we add some positive test cases for symbols?
Especially `"."` would be an interesting edge case (Matching only for the style symbols although all others start with `maybe(".")`)